source/core/lighting/photons.cpp

```diff
@@ -2280,18 +2280,21 @@ void PhotonGatherer::gatherPhotonsRec(int start, int end)
 
     // check this photon
 
-    // find distance from pt
-    ptToPhoton = - (*pt_s) + Vector3d(photon->Loc);
-    // all distances are squared
-    dx = ptToPhoton[X]*ptToPhoton[X];
-    dy = ptToPhoton[Y]*ptToPhoton[Y];
-    dz = ptToPhoton[Z]*ptToPhoton[Z];
+    // find distance in DimToUse from pt
+    d = photon->Loc[DimToUse] - (*pt_s)[DimToUse];
+    d *= d;
 
-    if (!(  ((dx>dmax_s) && ((DimToUse)==X)) ||
-            ((dy>dmax_s) && ((DimToUse)==Y)) ||
-            ((dz>dmax_s) && ((DimToUse)==Z)) ))
+    if (d<dmax_s)
     {
-        // it fits manhatten distance - maybe we can use this photon
+        // it fits manhatten, DimToUse distance - maybe we can use this photon
+
+        dx = photon->Loc[X] - (*pt_s)[X];
+        dy = photon->Loc[Y] - (*pt_s)[Y];
+        dz = photon->Loc[Z] - (*pt_s)[Z];
+        ptToPhoton = Vector3d(dx,dy,dz);
+        dx *= dx;
+        dy *= dy;
+        dz *= dz;
```

Comment:
Not got the code up and I'm shortly otherwise busy for much of the day. 

My recollection is the dx, dy and dz values are used somewhat later, but only conditionally. 

I'll look too at using Sqr() which I'd not done and look at better var names which is a good idea. 

The form settled upon though is part of the optimization. I didn't try to dig into the assembly code generated, but what seemed to be happening with more concise code was that the promotions from single to double tangled in the use of Vector3d  (it's inlined) was preventing some other compiler optimization. The more concise forms were slower and I ended up with what you see after being frustrated that code which should have been significantly faster wasn't after re-coding.  

I was only looking at the gnu compiler, but clang was similarly faster with the end result. 

I'll think on your comments more today. It will be late tonight - or more likely tomorrow before I can get the code up again and try things.